### PR TITLE
[Train] LightningTrainer always resumes from the latest AIR checkpoint during restoration.

### LINKED ISSUE #35613: [Train] LightningTrainer failed to auto-restore if `ckpt_path` provided in `LightningConfigbuilder.fit_param()`
### What happened + What you expected to happen

If users provide a `ckpt_path` in `LightningConfigBuilder.fit_param()`, when resuming training from a failed run, LightningTrainer will always resume training from that `ckpt_path` again, instead of the latest AIR checkpoint.

Code: https://github.com/ray-project/ray/blob/0fd06ad6fd54045e038f58ff75abd72c7adb2511/python/ray/train/lightning/lightning_trainer.py#L543-L549

Instead, we should always restore from the latest AIR checkpoint, if we can get it from the session, regardless of whether the user provided a `ckpt_path` or not.

### Versions / Dependencies

master

### Reproduction script

```python
lightning_config = (
    LightningConfigBuilder()
    .module(
        ...
    )
    .trainer(
        ...
    )
    .fit_params(
        train_dataloaders=train_loader,
        val_dataloaders=val_loader,
        ckpt_path="YOUR_CKPT_PATH"
    )
    .build()
)
```

Together with a FailureConfig
```python
run_config = RunConfig(
    ...,
    failure_config=FailureConfig(max_failures=3), 
)
```

### Issue Severity

High: It blocks me from completing my task.

### PR DESCRIPTION
**Pull request by woshiyyya**
## Why are these changes needed?

If users provide a `ckpt_path` in LightningConfigBuilder.fit_param(), when resuming training from a failed run, LightningTrainer will always resume training from `ckpt_path` again, instead of the latest AIR checkpoint. In that case, the training won't proceed after restored from failures.

Code: https://github.com/ray-project/ray/blob/0fd06ad6fd54045e038f58ff75abd72c7adb2511/python/ray/train/lightning/lightning_trainer.py#L543-L549

We should always restore from the latest checkpoint from `session.get_checkpoint()`, otherwise, the training re-start from the beginning after restoration.

Details in #35613.

## Related issue number

Close #35613 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by ArturNiederfahrenhorst** [comment]
@woshiyyya  Please make sure to also create the cherry pick PR yourself after this is merged!

File: python/ray/train/lightning/lightning_trainer.py [hunk 1]
```diff
@@ -541,7 +541,7 @@ def _lightning_train_loop_per_worker(config):
 
     # Restore from a previously failed run
     checkpoint = session.get_checkpoint()
-    if checkpoint and "ckpt_path" not in trainer_fit_params:
+    if checkpoint:
```
Comment (gjoliver) [inline python/ray/train/lightning/lightning_trainer.py:544]: we basically ignores what is in ckpt_path and always take checkpoint?
should we add a warning message here before we overwrite the config?
actually why would somebody specify both in this case?

Comment (woshiyyya) [reply python/ray/train/lightning/lightning_trainer.py:544]: No, we only ignore the `ckpt_path` when we are restoring from a failed run(`session.get_checkpoint() is not None`). If we start from scratch, we will still start from `ckpt_path`.

The use case is:
if a user has a checkpoint, and they want to resume training from that checkpoint, and they failed during training and wish to resume training from the latest one. Then we should ignore `ckpt_path` after restoration.

Comment (gjoliver) [reply python/ray/train/lightning/lightning_trainer.py:544]: yeah, that makes a lot of sense.
can we print a message here though, so we know this is happening in the logs.

**Review by gjoliver — APPROVED** [review]
ok, just 1 nit.
thanks.

File: python/ray/train/lightning/lightning_trainer.py [hunk 2]
```diff
@@ -541,7 +541,15 @@ def _lightning_train_loop_per_worker(config):
...
+        logger.info("Restoring LightningTrainer from a failed run. ")
```
Comment (matthewdeng) [inline python/ray/train/lightning/lightning_trainer.py:545]: Actually is this always true? What happens if the user passes in an AIR checkpoint?

```
LightningTrainer(..., resume_from_checkpoint=some_checkpoint)
```

Comment (woshiyyya) [reply python/ray/train/lightning/lightning_trainer.py]: It might not be a failed run. Updated the log message.

**Review by matthewdeng — APPROVED** [review]

**Pull request merged**